src/main/scala/beam/agentsim/agents/ridehail/RideHailModifyPassengerScheduleManager.scala

```diff
@@ -143,19 +143,27 @@ class RideHailModifyPassengerScheduleManager(
       case _ =>
         throw new RuntimeException("Should not attempt to send completion when doing single reservations")
     }
-    if (allTriggersInWave.nonEmpty)
-      rideHailManager.log.debug(
-        "Earliest tick in triggers to schedule is {} and latest is {}",
-        allTriggersInWave.map(_.trigger.tick).min,
-        allTriggersInWave.map(_.trigger.tick).max
-      )
+
+    if (allTriggersInWave.nonEmpty) {
+      debugAllTriggersInWave()
+    }
     scheduler.tell(
       CompletionNotice(triggerId, allTriggersInWave :+ ScheduleTrigger(timerTrigger, rideHailManagerRef)),
       rideHailManagerRef
     )
     allTriggersInWave = Vector()
   }
 
+  @SuppressWarnings(Array("UnsafeTraversableMethods"))
+  @inline
+  private def debugAllTriggersInWave(): Unit = {
+    if (rideHailManager.log.isDebugEnabled) {
+      val min = allTriggersInWave.map(_.trigger.tick).min
+      val max = allTriggersInWave.map(_.trigger.tick).max
+      rideHailManager.log.debug("Earliest tick in triggers to schedule is {} and latest is {}", min, max)
```

Comment:
We should double check that this is using the same logging as elsewhere, where the if isenabled is already built in (akka logger I think?)

Reply:
`isEnabled` is already implemented in all the logs system. But it will avoid only string objects creation. It won't avoid the calculation of `min`/ `max` that are expensive operations

Reply:
true - wasn't paying attention to those being done outside of the string